spark_client/utils.py

```diff
@@ -259,3 +274,49 @@ def add_deploy_arguments(parser):
         help="Deployment mode for job submission. Default is 'client'.",
     )
     return parser
+
+
+def get_driver_host_conf() -> Any:
```

Comment:
**thought** I've been thinking a bit about this. I'm starting to think that the best place for this would be the SparkInterface, when merging configurations. If we have a valid default (because of the auto-inference) we place that in the merged properties file, before reading the env property file and the one provided by the user. 

We then raise an error if the spark.driver.host was not in the merged properties OR in the extra_confs.